mapper/src/main/java/com/ibm/mapper/model/algorithms/FrodoKEM.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SonarQube Cryptography Plugin
+ * Copyright (C) 2024 IBM
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ibm.mapper.model.algorithms;
+
+import com.ibm.mapper.model.Algorithm;
+import com.ibm.mapper.model.KeyEncapsulationMechanism;
+import com.ibm.mapper.utils.DetectionLocation;
+import javax.annotation.Nonnull;
+
+public class FrodoKEM extends Algorithm implements KeyEncapsulationMechanism {
+    // https://frodokem.org/files/FrodoKEM-specification-20210604.pdf
+
+    private static final String NAME = "FrodoKEM";
+
+    public FrodoKEM(@Nonnull DetectionLocation detectionLocation) {
+        super(NAME, KeyEncapsulationMechanism.class, detectionLocation);
```

Comment:
Add OID and Nist security level here, especially important for quantum safe algorithms. As it seems the OID has to be generated using the enrichment, since it depends on multiple `INode`'s.

> FrodoKEM Variants and OIDs
> FrodoKEM has several variants, each with its own Object Identifier (OID): AES-based variants:
>
>    FrodoKEM-640-AES: 1.3.6.1.4.1.25258.1.15.1
>    FrodoKEM-976-AES: 1.3.6.1.4.1.25258.1.15.2
>    FrodoKEM-1344-AES: 1.3.6.1.4.1.25258.1.15.3
>
> SHAKE-based variants:
> 
>    FrodoKEM-640-SHAKE: 1.3.6.1.4.1.25258.1.14.1
>    FrodoKEM-976-SHAKE: 1.3.6.1.4.1.25258.1.14.2
>    FrodoKEM-1344-SHAKE: 1.3.6.1.4.1.25258.1.14.3
>
> Security Levels
> The different variants target different NIST security levels:
> 
>   FrodoKEM-640: Level 1 (comparable to AES-128)
>   FrodoKEM-976: Level 3 (comparable to AES-192)
>   FrodoKEM-1344: Level 5 (comparable to AES-256)

Reply:
Will be done later, as I first need to add detection rules for [FrodoParameters](https://downloads.bouncycastle.org/java/docs/bcprov-jdk18on-javadoc/org/bouncycastle/pqc/crypto/frodo/FrodoParameters.html)